src/main/java/com/epam/lowcost/DAO/PlaneDAOImpl.java

```diff
@@ -0,0 +1,104 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlaneDAOImpl implements PlaneDAO {
+
+    private DataSource dataSource;
+
+    public PlaneDAOImpl(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public List<Plane> getAllPlanes() {
+        List<Plane> allPlanes = new ArrayList<>();
+        try (Connection conn = dataSource.getConnection();
+             Statement stm = conn.createStatement();
+             ResultSet rs = stm.executeQuery("SELECT * FROM PLANES")) {
+            while (rs.next()) {
+                allPlanes.add(getPlanesFromDB(rs));
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return allPlanes;
+    }
+
+    @Override
+    public Plane getById(long planeId) {
+        Plane plane = null;
+        String sql = String.format("SELECT * FROM PLANES WHERE id='%d'", planeId);
+        try (Connection connection = dataSource.getConnection();
+             Statement stm = connection.createStatement();
+             ResultSet rs = stm.executeQuery(sql)) {
+            if (rs.next()) {
+                return getPlanesFromDB(rs);
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return plane;
+    }
+
+    @Override
+    public Plane addPlane(Plane plane) {
+        String sql = String.format(
+                "INSERT INTO PLANES (model, businessPlacesNumber, economPlacesNumber) " +
+                        "VALUES ('%s', %d, %d)",
+                plane.getModel(), plane.getBusinessPlacesNumber(), plane.getEconomPlacesNumber());
+        try (Connection connection = dataSource.getConnection();
+             Statement stm = connection.createStatement()
+        ) {
+            int insert = stm.executeUpdate(sql);
+            if (insert == 1) {
+                ResultSet rs = stm.executeQuery("SELECT * FROM PLANES");
+                rs.last();
+                long newId = rs.getLong("id");
+                plane.setId(newId);
+                return plane;
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        plane = null;
+        return plane;
+    }
+
+    @Override
+    public Plane updatePlane(Plane plane) {
+        String sql = String.format(
+                "UPDATE PLANES SET model='%s',businessPlacesNumber=%d,economPlacesNumber=%d WHERE id=%d",
+                plane.getModel(), plane.getBusinessPlacesNumber(), plane.getEconomPlacesNumber(), plane.getId());
+        try (Connection conn = dataSource.getConnection();
+             Statement stm = conn.createStatement()) {
+            int update = stm.executeUpdate(sql);
+            if (update == 1) {
+                return plane;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return plane;
+    }
+
+    private Plane getPlanesFromDB(ResultSet rs) throws SQLException {
```

Comment:
I think that this name is not appropriate. Maybe more suitable `extract...`... I'm sure you can create new good name)

Reply:
Changed method name